Projections.bean discards aliases required for sorting

```java
final List<Expression<?>> list = new ArrayList<>();
final NumberPath<Double> avgAmountPath = Expressions.numberPath(Double.class, "avgAmount");
list.add(QEntity.entity.amount.avg().as(avgAmountPath));
jpaQuery.groupBy(QEntity.entity.something);
jpaQuery.orderBy(avgAmountPath.asc());

// this throws exception: Caused by: com.mysql.jdbc.exceptions.jdbc4.MySQLSyntaxErrorException: Unknown column 'avgAmount' in 'order clause'
jpaQuery.list(Projections.bean(MyResult.class, list.toArray(new Expression[0])));

// this is ok
jpaQuery.list(list.toArray(new Expression[0]));
```

--- a/querydsl-core/src/main/java/com/querydsl/core/types/QBean.java
+++ b/querydsl-core/src/main/java/com/querydsl/core/types/QBean.java
@@ -27,6 +27,7 @@ import java.util.Map;
 import com.google.common.collect.ImmutableList;
 import com.google.common.collect.ImmutableMap;
 import com.google.common.primitives.Primitives;
+import com.querydsl.core.group.GroupExpression;
 
 /**
  * {@code QBean} is a JavaBean populating projection type
@@ -61,7 +62,11 @@ public class QBean<T> extends FactoryExpressionBase<T> {
                 Operation<?> operation = (Operation<?>) expr;
                 if (operation.getOperator() == Ops.ALIAS && operation.getArg(1) instanceof Path<?>) {
                     Path<?> path = (Path<?>) operation.getArg(1);
-                    rv.put(path.getMetadata().getName(), operation.getArg(0));
+                    if (isCompoundExpression(operation.getArg(0))) {
+                        rv.put(path.getMetadata().getName(), operation.getArg(0));
+                    } else {
+                        rv.put(path.getMetadata().getName(), operation);
+                    }
                 } else {
                     throw new IllegalArgumentException("Unsupported expression " + expr);
                 }
@@ -71,6 +76,10 @@ public class QBean<T> extends FactoryExpressionBase<T> {
             }
         }
         return rv.build();
+    }
+
+    private static boolean isCompoundExpression(Expression<?> expr) {
+        return expr instanceof FactoryExpression || expr instanceof GroupExpression;
     }
 
     private static Class<?> normalize(Class<?> cl) {

--- a/querydsl-core/src/test/java/com/querydsl/core/types/QBeanTest.java
+++ b/querydsl-core/src/test/java/com/querydsl/core/types/QBeanTest.java
@@ -180,4 +180,10 @@ public class QBeanTest {
         assertFalse(bean.equals(bean.skipNulls()));
     }
 
+    @Test
+    public void Alias() {
+        QBean<Entity> beanProjection = new QBean<Entity>(Entity.class, name.as("name2"));
+        assertEquals(name.as("name2"), beanProjection.getArgs().get(0));
+    }
+
 }

